gateway/gateway.go

```diff
@@ -185,18 +200,15 @@ func (g *Gateway) Close() error {
 // Reconnect tries to reconnect forever. It will resume the connection if
 // possible. If an Invalid Session is received, it will start a fresh one.
 func (g *Gateway) Reconnect() {
-	for {
-		if err := g.ReconnectCtx(context.Background()); err != nil {
-			g.ErrorLog(err)
-		} else {
-			return
-		}
-	}
+	ctx, cancel := context.WithTimeout(context.Background(), g.WSTimeout)
+	defer cancel()
+
+	g.ReconnectCtx(ctx)
```

Comment:
`ReconnectCtx` should be called with a `context.Background()`, as we want to default to endlessly trying to reconnect.